src/libsyntax/parse/lexer/mod.rs

```diff
@@ -1873,13 +1885,14 @@ fn is_block_doc_comment(s: &str) -> bool {
     res
 }
 
+/// Determine whether `c` is a valid start for an ident.
 fn ident_start(c: Option<char>) -> bool {
     let c = match c {
         Some(c) => c,
         None => return false,
     };
 
-    (c >= 'a' && c <= 'z') || (c >= 'A' && c <= 'Z') || c == '_' || (c > '\x7f' && c.is_xid_start())
+    (c.is_alphabetic() || c == '_' || (c > '\x7f' && c.is_xid_start()))
```

Comment:
The point of this code and the code in `ident_continue` below was optimization for the common case.
In addition to that `is_alphabetic` is different from `is_xid_start`, so this is also a change in behavior.
I'd remove these changes.

Reply:
I thought that both `is_alphabetic` and `is_xid_start` had those optimizations as well, but it seems like it's not the case. I'll change it.